File: demo/src/main/java/com/kesco/adk/moko/slideback/demo/MainActivity.java
package com.kesco.adk.moko.slideback.demo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.kesco.adk.moko.slideback.Slider;

/**
 * Created by kesco on 15/7/24.
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Slider.INSTANCE$.attach(this);

        findViewById(R.id.btn_simulate).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                KeyEvent kDown = new KeyEvent(KeyEvent.ACTION_DOWN, KeyEvent.KEYCODE_BACK);
//                dispatchKeyEvent(kDown);
//                KeyEvent kUp = new KeyEvent(KeyEvent.ACTION_UP, KeyEvent.KEYCODE_BACK);
//                dispatchKeyEvent(kUp);
                Intent intent = new Intent(MainActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: showcase/src/main/kotlin/com/kesco/adk/sliding/demo/MainActivity.kt
package com.kesco.adk.sliding.demo

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button

class MainActivity : AppCompatActivity() {
  private lateinit var _btnNew: Button;

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    _btnNew = findViewById(R.id.btn_create) as Button
    _btnNew.setOnClickListener {
      val intent = Intent(this, DirectAttachActivity::class.java)
      startActivity(intent)
    }
  }
}
